dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java

```diff
@@ -2038,33 +2038,41 @@ boolean reportErrors(Result result, Collection<String> restrictedAccessAttempts,
             }
         }
 
-        List<String> unresolvedVersionless = new ArrayList<>();
-        for (Map.Entry<String, String> versionlessResolved : result.getVersionlessFeatures().entrySet()) {
-            if(versionlessResolved.getValue() == null){
-                String platformBaseName = featureRepository.getPlatformForVersionlessFeature(versionlessResolved.getKey());
-
-                boolean compatibleWithPlatform = true;
-                if(platformBaseName != null && !result.getResolvedPlatforms().isEmpty()){
-                    for(String resolvedPlat : result.getResolvedPlatforms()){
-                        if(resolvedPlat.toLowerCase().startsWith(platformBaseName.toLowerCase()) && 
-                            featureRepository.getVersionlessFeatureVersionForPlatform(versionlessResolved.getKey(), resolvedPlat) == null){
-
-                            Tr.error(tc, "INCOMPATIBLE_VERSIONLESS_FEATURE_WITH_PLATFORM", getFeatureName(versionlessResolved.getKey()), resolvedPlat);
-                            compatibleWithPlatform = false;
-                            break;
-                        }
-                    }
+        for (Map.Entry<String, String> versionlessResolvedEntry : result.getVersionlessFeatures().entrySet()) {
+            String versionlessResolved = versionlessResolvedEntry.getKey();
+            String versionedResolved = versionlessResolvedEntry.getValue();
+
+            if(versionedResolved != null){
+                continue; // Sucessfully resolved.  Nothing to check.
+            }
+
+            Set<String> platforms = featureRepository.getPlatformsForVersionlessFeature(versionlessResolved);
+            if ((platforms == null) || platforms.isEmpty()) {
+                continue; // Compatibility or linking features are not installed.  The image is incomplete.
+            }
+
+            ProvisioningFeatureDefinition compatibility = null;
+            for (String platform : platforms) {
+                compatibility = featureRepository.getCompatibilityFeature(platform);
+                if(compatibility != null){
+                    break;
                 }
+            }
+            if (compatibility == null) {
+                continue; // Compatibility features are not installed.  The image is incomplete.
+            }
 
-                if(compatibleWithPlatform){
-                    unresolvedVersionless.add(getFeatureName(versionlessResolved.getKey()));
+            String compatibilityBaseName = featureRepository.getFeatureBaseName(compatibility.getFeatureName());
+            for(String resolvedPlat : result.getResolvedPlatforms()){
+                String resolvedBaseName = featureRepository.getFeatureBaseName(featureRepository.getCompatibilityFeature(resolvedPlat).getFeatureName());
```

Comment:
Can you do a null check here?  Sorry but think the repo could be minified...  and not all the compat features are there?   this is what we ran into on the weekend